src/charm.py

```diff
@@ -301,16 +302,60 @@ def _prometheus_config(self):
 
         return yaml.dump(scrape_config)
 
+    def _get_image_tag(self, api_tags=None) -> str:
+        """Get most recent patch of major and minor Prometheus version
+        """
+        # 1. get valid tags from URL (or directly from input)
+        # 2. clean tags data structure and filter out release candidates "rc"
+        # 3. filter out tags that don't match our major/minor specification
+        config = self.model.config
+        if not api_tags:
+            api_tags_resp = requests.get(config['image-tags-url'])
```

Comment:
The concepts described here feel like something we want to encapsulate in a reusable Component, since we'll want the same behavior for Grafana, and Mattermost, and all the other things that use images.

Reply:
Agreed. I was thinking about how to do that - but it seems like every repo has it's own tagging structure which would make picking the latest patch for any repo pretty complicated.

Maybe there is a way to do this, but I think it would have to happen when we have ROCKs. I could be wrong and happy for suggestions.